packages/cli/src/commands/build.ts

```diff
@@ -409,6 +409,9 @@ export async function build({
     process.off('SIGTERM', handler);
     process.off('SIGQUIT', handler);
 
+    const filteredWriteIpfsUrl = cliSettings.writeIpfsUrl?.replace(RegExp(/:[a-zA-Z0-9]+@/), `:${'*'.repeat(20)}@`);
```

Comment:
This Regex does not work when the password has special characters like `!-_`, nevertheless, it would be easier to use the native Node's URL class, e.g.:
```typescript
const filterUrlPassword = (uri: sting) => {
  const res = new URL(uri);
  res.password = '*'.repeat(10);
  return res.string;
}

// ...

const filteredWriteIpfsUrl = filterUrlPassword(cliSettings.writeIpfsUrl)
```

Reply:
Thanks for the review! I've refactored the code to use the internal URL lib.